sklearn/utils/estimator_checks.py

```diff
@@ -1248,6 +1259,22 @@ def check_estimators_nan_inf(name, estimator_orig):
                     raise AssertionError(error_string_transform, estimator)
 
 
+@ignore_warnings
+def check_nonsquare_error(name, estimator_orig):
+    """Test that error is thrown when non-square data provided"""
+
+    if name != "KernelCenterer":
```

Comment:
Why are we skipping the test for the other ones?

Reply:
@adrinjalali. I was not sure if the test was valid for other pairwise estimators as it was causing unit tests to fail.  I was able to fix it for precomputed SVM by just changing `estimator.fit(X) -> estimator.fit(X, y)`.  

However, precomputed KNeighborRegressor failed to raise a value error on non-square data.  I tried to add a check for non-square data, but it seems other tests of KNR require non-square precomputed input matrix.  The doc says the data should be n_sample x n_sample.  What is correct? Thanks.